virtual_rainforest/models/litter/litter_pools.py

```diff
@@ -0,0 +1,162 @@
+
+from virtual_rainforest.models.litter.constants import LitterConsts
+
+# TODO - At the moment this module does not use litter chemistry (relative lignin
+# content) at all. We need to decide how we handle this and adjust the below functions
+# to use this at some point.
+
+
+def calculate_litter_pool_updates(
+    surface_temp: NDArray[np.float32],
+    litter_pool_above_metabolic: NDArray[np.float32],
+    litter_pool_above_structural: NDArray[np.float32],
+    update_interval: float,
+    constants: LitterConsts,
+) -> dict[str, DataArray]:
+    """Calculate updates for all litter pools.
+
+    Args:
+        surface_temp: Temperature of soil surface, which is assumed to be the same
+            temperature of the above ground litter [C]
+        litter_pool_above_metabolic: Above ground metabolic litter pool [kg C m^-2]
+        litter_pool_above_structural: Above ground structural litter pool [kg C m^-2]
+        update_interval: Interval that the litter pools are being updated for [days]
+        constants: Set of constants for the litter model
+
+    Returns:
+        The new value for each of the litter pools
+    """
+
+    # Calculate temperature factor for the above ground litter layers
+    temperature_factor_above = calculate_temperature_effect_on_litter_decomp(
+        temperature=surface_temp,
+        reference_temp=constants.litter_decomp_reference_temp,
+        offset_temp=constants.litter_decomp_offset_temp,
+        temp_response=constants.litter_decomp_temp_response,
+    )
+
+    # Calculate the pool decay rates
+    metabolic_above_decay = calculate_litter_decay_metabolic_above(
+        temperature_factor_above,
+        litter_pool_above_metabolic,
+        litter_decay_coefficient=constants.litter_decay_constant_metabolic_above,
+    )
+    structural_above_decay = calculate_litter_decay_structural_above(
+        temperature_factor_above,
+        litter_pool_above_structural,
+        litter_decay_coefficient=constants.litter_decay_constant_structural_above,
+    )
+
+    # Combine with input rates and multiple by update time to find overall changes
+    change_in_metabolic_above = (
+        constants.litter_input_to_metabolic_above - metabolic_above_decay
+    ) * update_interval
+    change_in_structural_above = (
+        constants.litter_input_to_structural_above - structural_above_decay
+    ) * update_interval
+
+    # Construct dictionary of data arrays to return
+    new_litter_pools = {}
+    new_litter_pools["litter_pool_above_metabolic"] = DataArray(
+        litter_pool_above_metabolic + change_in_metabolic_above, dims="cell_id"
+    )
+    new_litter_pools["litter_pool_above_structural"] = DataArray(
+        litter_pool_above_structural + change_in_structural_above, dims="cell_id"
+    )
+
+    return new_litter_pools
+
+
+def calculate_temperature_effect_on_litter_decomp(
+    temperature: NDArray[np.float32],
+    reference_temp: float,
+    offset_temp: float,
+    temp_response: float,
```

Comment:
Should these default to the `LitterConsts` defaults?

```suggestion
    reference_temp: float = LitterConsts.litter_decomp_reference_temp,
    offset_temp: float = LitterConsts.litter_decomp_offset_temp,
    temp_response: float = = LitterConsts.litter_decomp_temp_response,
```

Reply:
I moved away from doing that with the new constants system. It seemed a bit easy for my liking to forget to give all the arguments to the function and then inadvertently supply the default as defined in as defined in the original data class (apresuming that `LitterConsts.constant_name` supplies that default). An oversight like this would mean that sensitivity analysis wasn't actually changing the parameter that it was claiming to change

Reply:
This might work different with a constants registry though?

Reply:
I sort of see that, but it does seem extremely strange to have an explicit set of defaults for a function like this and then not to connect those up so that they get used in the function. I need to think about this one but it doesn't feel right.

Reply:
The alternative feels strange as well though as you are providing default values that should never be used, unless the functions are called individually outside the main simulation flow. I can't see this happening much for this level of function (ones that calculate individual variables for a specific model)? Even for plotting simulation results you probably want to make sure you supply the constants used by the simulation, rather than the defaults

Reply:
Yeah think this works best as an issue, as I think we need to decide on a consistent approach to this across models and then document it